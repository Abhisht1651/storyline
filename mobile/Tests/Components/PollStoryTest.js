// @flow
import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import PollStory from '../../App/Components/PollStory'

describe('PollStory', () => {
  it('PollStory component renders correctly', () => {
    const wrapper = shallow(<PollStory />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the Text Component', () => {
    const wrapper = shallow(<PollStory />)
    expect(wrapper.find('Text')).toHaveLength(10)
  })
  it('should render the Image Component', () => {
    const wrapper = shallow(<PollStory />)
    expect(wrapper.find('Image')).toHaveLength(9)
  })
})
