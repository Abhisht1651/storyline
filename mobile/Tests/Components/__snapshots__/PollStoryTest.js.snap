// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollStory PollStory component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderBottomLeftRadius": 30,
      "borderBottomRightRadius": 30,
      "borderColor": "#E6ECED",
      "borderStyle": "solid",
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "borderWidth": 1,
      "minHeight": 300,
      "minWidth": 300,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 28,
        "width": "88%",
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "height": 24,
          "width": 28,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Regular",
          "fontSize": 32,
          "letterSpacing": -0.12,
          "marginLeft": 8,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Poll
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 8,
        "width": "88%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Medium",
          "fontSize": 36,
          "letterSpacing": -0.11,
          "lineHeight": 44,
          "opacity": 1,
          "textAlign": "left",
        }
      }
    >
      Who are your favorites for the World Cup 2020?
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 24,
        "paddingBottom": 10,
        "width": "88%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginLeft": 8,
          }
        }
      >
        <Image
          key="0"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "marginLeft": -16,
              "opacity": 1,
              "width": 56,
            }
          }
        />
        <Image
          key="1"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "marginLeft": -16,
              "opacity": 1,
              "width": 56,
            }
          }
        />
        <Image
          key="2"
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "borderRadius": 28,
              "height": 56,
              "marginLeft": -16,
              "opacity": 1,
              "width": 56,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#082366",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 28,
              "letterSpacing": -0.1,
              "opacity": 0.5,
              "textAlign": "left",
            }
          }
        >
          54
          K votes
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 34,
              "width": 34,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "marginLeft": 6,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#002468",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 24,
              "letterSpacing": -0.07,
              "opacity": 0.5,
              "textAlign": "right",
            }
          }
        >
          2 hours left
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 29,
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "40%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#8F8FAF",
            "borderRadius": 24,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 28,
              "letterSpacing": 0,
              "marginTop": 2,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          India
        </Text>
      </View>
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "40%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#8F8FAF",
            "borderRadius": 24,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 28,
              "letterSpacing": 0,
              "marginTop": 2,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          West Indies
        </Text>
      </View>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 29,
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "40%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#8F8FAF",
            "borderRadius": 24,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 28,
              "letterSpacing": 0,
              "marginTop": 2,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          Australia
        </Text>
      </View>
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "width": "40%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#8F8FAF",
            "borderRadius": 24,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Light",
              "fontSize": 28,
              "letterSpacing": 0,
              "marginTop": 2,
              "opacity": 1,
              "textAlign": "left",
            }
          }
        >
          England
        </Text>
      </View>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "marginVertical": 40,
      }
    }
  >
    <MediumThinnerButton
      onPress={[Function]}
      outline={true}
      style={Object {}}
      title="Vote"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#E6ECED",
        "borderBottomWidth": 1,
        "borderTopColor": "#E6ECED",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 37.5,
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Light",
            "fontSize": 28,
            "letterSpacing": 0,
            "opacity": 1,
            "paddingLeft": "7.1%",
            "textAlign": "left",
          }
        }
      >
        154
        K Comments
      </Text>
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "paddingRight": "7.1%",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 26,
            "marginRight": 5,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#8F8FAF",
            "fontFamily": "BananaGrotesk-Regular",
            "fontSize": 28,
            "letterSpacing": -0.08,
            "opacity": 1,
            "textAlign": "left",
          }
        }
      >
        Comment
      </Text>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginVertical": 40,
        "width": "82.857%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 36,
            "opacity": 1,
            "width": 130,
          }
        }
      />
    </ForwardRef>
  </View>
</View>
`;
